torch/csrc/stable/stableivalue_conversions.h

```diff
@@ -200,6 +235,30 @@ struct FromImpl<torch::stable::Tensor> {
   }
 };
 
+// Specialization for torch::stable::Device => StableIValue
+// Pack the device type and index into a StableIValue in a platform-independent
+// format. We use the shim representation for DeviceType (int32_t) for ABI
+// stability. StableIValue layout: DeviceType (shim int32_t) in lower 32 bits,
```

Comment:
I didn't use the default `from` because I worry that struct padding might differ across systems

I reversed the order of DeviceType and DeviceIndex because DeviceIndex can be negative (putting it in front gives it the sign bit)